### ISSUE TITLE
Ickle fulltext query not working with query parameters

### ISSUE THREAD

**Issue by martin-kanis** [keycloak/keycloak#9292]
### Describe the bug

I have an Ickle fulltext query on analyzed field and it doesn't handle query parameters. 
```
QueryFactory queryFactory = Search.getQueryFactory(remoteCache);
private final Map<String, String> parameters;
parameters.put("clientId0", "test")
queryString = "FROM org.keycloak.models.map.storage.hotrod.HotRodClientEntity c WHERE (c.clientId : :clientId0)";
Query<V> query = queryFactory.create(queryString);
query.setParameters(parameters);
....
```

No result is found in the remote cache. 

### Version

16

### Anything else?

Issue at Infinispan side: https://issues.redhat.com/browse/ISPN-13499

**Comment by ahus1** [comment]
Hi @martin-kanis - I updated the labels of this issue so that it is no longer in triage status. Please let me know if the labels should be tuned differently.

**Comment by martin-kanis** [comment]
The bug is fixed in Infinispan 14.0.0.Dev03. In order to test it, I needed to upgrade Infinispan version in Keycloak and start Infinispan server in version 14.0.0.Dev03.
Required tweaks, mostly related to creation of caches and removing workarounds on Keycloak side, can be found here: https://github.com/martin-kanis/keycloak/commit/040154be766a7432834a378f9f00e8292fd70f38
Note these are just quick and dirty fixes to verify the bug is fixed.

**Comment by ahus1** [comment]
As announced to the community in [this blog post](https://www.keycloak.org/2023/10/map-store-removal), we are no longer developing the Map Store, and it will be removed from Keycloak's code base. Follow #24111 for the removal steps.
